src/UserModule/repository/user.repository.ts

```diff
@@ -11,4 +11,13 @@ export class UserRepository extends Repository<User> {
   async findByIdWithCertificates(id: string) {
     return this.findOneOrFail(id, { relations: ['certificates'] });
   }
+  
+  public async getCertificateByUser(userId): Promise<any[]> {
+    
+    return createQueryBuilder("user", "user")
```

Comment:
Invés de usar o createQueryBuilder, você pode usar as options do find no RepositoryAPI

exemplo:
```javascript
userRepository.find({ 
    join: {
        alias: "user",
        leftJoinAndSelect: {
            profile: "user.profile",
            photo: "user.photos",
            video: "user.videos"
        }
    }
});
```

documentação: https://github.com/typeorm/typeorm/blob/master/docs/find-options.md

Reply:
Olá Davide, tudo certo?
é impeditivo neste momento usar createQueryBuilder?
tanto eu quanto o Thiago não temos experiência com typeorm. 
conseguimos depois de muito bater a cabeça implementar com createQueryBuilder.
até tentamos implementar seguindo a documentação que indicou mas sem sucesso. talvez falte um pouco mais de tempo para estudar e implementar. minha preocupação e com o prazo. esta tarefa se estendeu um pouco mais do que planejamos. o objetivo da tarefa foi cumprido e o mais importante o código está legível e de fácil entendimento para pessoas com pouco experiência que é o caso do Thiago. 

se puder auxiliar o Thiago para implementar da forma que acha mais correto (tecnicamente falando) será de bom grado e uma ótima oportunidade de aprendizagem.

Reply:
É que a repository api é mais próximo das entidades do que do SQL e o createQueryBuilder é mais próximo de SQL, então eu gostaria de manter o que for possível usando Repository API. Podemos aceitar assim e depois estudar os casos que estão usando createQueryBuilder se podemos mudar para Repository API, não quero prejudicar a PR de vocês por causa disso

Reply:
Boa Davide. junto com o Thiago iremos estudar essa mudança.